### ISSUE TITLE
PLEASE INFORM: Is reactionic still being maintained?

### ISSUE THREAD

**Issue by ghost** [reactionic/reactionic#75]
Looking through the issues, many recent ones remain unanswered. Is this library still being maintained?

Aka, is it recommended that we use it to write a large scale production hybrid mobile app?

If not, what are the alternatives?

**Comment by pors** [comment]
It is no longer maintained by me, but I'm still looking for someone to take over.

I would recommend React-Native.

**Comment by gabrielseco** [comment]
Maybe we can edit  the README.MD that reactionic is not longer maintained, possible alternatives and we're looking for maintainers

**Comment by pors** [comment]
I asked the Ionic organization. Let's see if they are interested. If not
then yes, do what you propose.

**Comment by chemitaxis** [comment]
Any news? Thanks!

**Comment by alicerocheman** [comment]
Hi, I just saw that this is no longer maintained. Here's my 2 cents.
React Native is no reason to drop react ionic. React Native doesn't allow cross coding for mobile devices AND desktop. 
Personnaly I need a hybrid solution that allows this with react, and believe me it's not that easy to find. I was so glad to find react ionic, thought my search was finally over, and... well I can't really make my company's app depend on a library that's not maintained...
So I'm back to square 1. 
I understand that you might have dropped this for many reasons, but just so you know, there's an audience and we can't all use react native.
On that note, I hope someone will pick this up and start maintaining this very soon!

Take care and thanks for your work :)

**Comment by pors** [comment]
I approached the Ionic foundation but they are not interested in maintaining reactionic. If anyone else is, please let me know and I'll add you to the organization.

**Comment by johnwillyn** [comment]
Any reason why Ionic is not interested?  Seems much cleaner than some of the Angular code I have seen in the past, but I have not looked at their latest.

I (we) may be interested in picking this up.  I don't feel react native is quite ready, and is probably going to take a couple of years before we have a mature ecosystem like that around Cordova.

Other than picking this up and exploring, what are the immediate issues and milestones that need to be addressed?

JohnL

**Comment by gabrielseco** [comment]
Maybe @pors can give you more ideas than me but before we stop coding there were warnings with React 15 because we we were passing unknown props to some components. 

You should convert the React.createClass components to es6 components I think because React.createClass is getting deprecated in React 16.

If you're interested in doing a ionic 2 version, I think you'll need to do a complete rewrite and you could do it in a separated branch.

About ionic not being interested, there's something they are talking about called stenciljs (not out yet), I think there are webcomponents that work with angularjs angular 2 or react out of the box.

https://www.npmjs.com/package/@stencil/core

If you read the readme you can see the thing they are trying to implement has caught ideas from React

**Comment by pors** [comment]
@johnwillyn this is what the Ionic CEO responded:

> Thanks for the email. You are correct that we were and remain impressed by the project :)
> 
> Right now we're focused on Ionic 2 final so we won't be able to take this over, but I just forked the repo into our org to be safe :)
> 
> I'll be in touch should we decide it's something we can take on.

That was almost a year ago, so I think that is not going to happen :)

About picking this up: that would be great! I agree with @gabrielseco, if you want to continue with Ionic 1 (the CSS of Ionic 1) I would start upgrading all dependencies and fix what breaks. 

Supporting Ionic 2 needs a lot of work and maybe starting from scratch is best in that case (but I haven't looked into it, so maybe there's lots of reusable code).

**Comment by johnwillyn** [comment]
Thanks @pors.  I have not looked at Ionic 2 in detail, but I am familiar with the bulk of Ionic 1, so I would probably start there and update the React to current.  

I will probably give this some effort this week and see where it takes me.

John

**Comment by johnwillyn** [comment]
I spent the morning updating the reactionic-kitchensink demo app dependencies to the current react-ionic, etc.  

Many of the issues were related to updating the webpack dev dependency (in the end).

The one issue I don't know how to resolve is the inconsistency of Meteor's handling of this line in app/client/styles/app.sccs:

@import "{}/node_modules/reactionic/dist/scss/styles/reactionic.scss";

I had to follow the suggested change from @gabrielseco for this change.

I still have the warnings about the deprecated React features for createClass, but it appears that the Facebook guys have a code migration script for updating all of the components for that change.

I do like this framework as I have been a user of Ionic 1 for a while, and I think their app -> layout -> content approach (with runtime definable content) suits an app I working on, so I will give this a bit more time and effort to see if I can get all of this up on React 15.5 without warnings.

**Comment by pors** [comment]
@johnwillyn awesome! Let me know if you would like access to the repo/organization.

**Comment by johnwillyn** [comment]
Just to be clear, I updated the reactionic-kitchensink demo build to move to reactionic 1.2.0-beta.0 and a few of the build dependencies there that were outdated as well.

I did not update the core react, react-router, react-dom, or transitions dependencies beyond the 15.0.2 versions, but that is the next step (I suggested above).

I will submit a PR for my changes to the demo app if someone is interested.
